flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -1607,6 +1613,103 @@ private void getActiveNotifications(Result result) {
+      return;
+    }
+    NotificationManager notificationManager =
+        (NotificationManager) applicationContext.getSystemService(Context.NOTIFICATION_SERVICE);
+    try {
+      Map<String, Object> arguments = call.arguments();
+      int id = (int) arguments.get("id");
+      String tag = (String) arguments.get("tag");
+
+      StatusBarNotification[] activeNotifications = notificationManager.getActiveNotifications();
+      Notification notification = null;
+      for (StatusBarNotification activeNotification : activeNotifications) {
+        if (activeNotification.getId() == id && (tag == null || activeNotification.getTag() == tag)) {
+          notification = activeNotification.getNotification();
+          break;
+        }
+      }
+
+      if (notification == null) {
+        result.success(null);
+        return;
+      }
+
+      NotificationCompat.MessagingStyle messagingStyle = NotificationCompat.MessagingStyle.extractMessagingStyleFromNotification(notification);
+      if (messagingStyle == null) {
+        result.success(null);
+        return;
+      }
+
+      HashMap<String, Object> stylePayload = new HashMap<>();
+      stylePayload.put("groupConversation", messagingStyle.isGroupConversation());
+      stylePayload.put("person", describePerson(messagingStyle.getUser()));
+      stylePayload.put("conversationTitle", messagingStyle.getConversationTitle());
+
+      List<Map<String, Object>> messagesPayload = new ArrayList<>();
+      for (NotificationCompat.MessagingStyle.Message msg : messagingStyle.getMessages()) {
+        Map<String, Object> msgPayload = new HashMap<>();
+        msgPayload.put("text", msg.getText());
+        msgPayload.put("timestamp", msg.getTimestamp());
+        msgPayload.put("person", describePerson(msg.getPerson()));
+        messagesPayload.add(msgPayload);
+      }
+      stylePayload.put("messages", messagesPayload);
+
+      result.success(stylePayload);
+    } catch (Throwable e) {
+      result.error(GET_ACTIVE_MESSAGING_STYLE_ERROR_CODE, e.getMessage(), e.getStackTrace());
+    }
+  }
+
+  private Map<String, Object> describePerson(Person person) {
+    if (person == null) {
+      return null;
+    }
+    Map<String, Object> payload = new HashMap<>();
+    payload.put("key", person.getKey());
+    payload.put("name", person.getName());
+    payload.put("uri", person.getUri());
+    payload.put("bot", person.isBot());
+    payload.put("important", person.isImportant());
+    payload.put("icon", describeIcon(person.getIcon()));
+    return payload;
+  }
+
+  private Map<String, Object> describeIcon(IconCompat icon) {
+    if (icon == null) {
+      return null;
+    }
+    IconSource source;
+    Object data;
+    switch (icon.getType()) {
+      case IconCompat.TYPE_RESOURCE:
+        source = IconSource.DrawableResource;
+        int resId = icon.getResId();
+        Context context = applicationContext;
+        assert(context.getResources().getResourceTypeName(resId) == DRAWABLE);
```

Comment:
These assertions don't work due to the way equality checks are done in Java and currently leads to exception whilst debugging. If you're using something like Android Studio, the IDE is able to suggest a fix on this i.e. use the `equals()` method